packages/core/src/Stores/Modules/Cashier/cashier-store.js

```diff
@@ -280,6 +289,49 @@ export default class CashierStore extends BaseStore {
             this.setSessionTimeout(false);
             this.setTimeoutCashierUrl();
         }
+        await BinarySocket.wait('get_account_status');
+        this.checkCashierLock();
+        if (this.active_container === this.config.deposit.container) {
+            this.checkDepositLock();
+        }
+        if (this.active_container === this.config.withdraw.container) {
+            this.checkWithdrawalLock();
+        }
+        if (this.active_container === this.config.account_transfer.container) {
+            this.checkTransferLock();
+        }
+    }
+
+    @action.bound
+    checkCashierLock() {
+        this.is_cashier_locked = this.root_store.client.account_status.status.some(
+            status => status === 'cashier_locked'
+        );
+    }
+
+    @action.bound
+    checkDepositLock() {
```

Comment:
can `checkDepositLock` `checkTransferLock` and `checkWithdrawalLock` be computed?

Also what about breaking down the condition a bit as it's quite the mouthful to read now? (for the other 2 as well)

```
        const { is_authentication_needed,  is_tnc_needed, is_financial_account, is_financial_information_incomplete, is_trading_experience_incomplete } = this.root_store.client

        this.is_deposit_locked =
            !!is_authentication_needed || is_tnc_needed ||
            (is_financial_account && 
                    (is_financial_information_incomplete || is_trading_experience_incomplete));